### PR TITLE
add REST scripts

### PR DESCRIPTION
**Pull request by jc-cb**
-add REST scripts

### CODE REVIEW

File: rest/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Exchange API REST Scripts
+
+This repository provides examples of common REST API endpoints to use with Coinbase Exchange.
```
Comment (rcbgr) [inline rest/README.md:3]: Add a link to the Exchange REST API docs

File: rest/exchange_create_conversion.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+SECRET_KEY = str(os.environ.get('SIGNING_KEY'))
+
+url = f'https://api.exchange.coinbase.com/conversions'
+
+timestamp = str(int(time.time()))
+method = 'POST'
+
+url_path = urlparse(url).path
+
+profile_id = 'profile_id'
```
Comment (rcbgr) [inline rest/exchange_create_conversion.py:29]: What does this do? Should this be an env var?

File: rest/exchange_create_crypto_address.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+import json, hmac, hashlib, time, requests, base64, os, sys
+from urllib.parse import urlparse
+
+API_KEY = str(os.environ.get('ACCESS_KEY'))
+PASSPHRASE = str(os.environ.get('PASSPHRASE'))
+SECRET_KEY = str(os.environ.get('SIGNING_KEY'))
+
+if len(sys.argv) != 2:
+    print("Error: One command-line argument is required.")
```
Comment (rcbgr) [inline rest/exchange_create_crypto_address.py:23]: This line is unnecessary. Just print the usage

File: rest/exchange_create_crypto_address.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    print("Usage: python exchange_create_crypto_address.py <account_id>")
```
Comment (rcbgr) [inline rest/exchange_create_crypto_address.py:24]: Use sys.exit("usage message") and then no need for sys.exit and print

File: rest/exchange_create_deposit_from_coinbase.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+# limitations under the License.
+
+import json, hmac, hashlib, time, requests, base64, os, sys
+from urllib.parse import urlparse
+
+API_KEY = str(os.environ.get('ACCESS_KEY'))
+PASSPHRASE = str(os.environ.get('PASSPHRASE'))
+SECRET_KEY = str(os.environ.get('SIGNING_KEY'))
+
+url = f'https://api.exchange.coinbase.com/deposits/coinbase-account'
```
Comment (rcbgr) [inline rest/exchange_create_deposit_from_coinbase.py:22]: There isn't any string interpolation

File: rest/exchange_create_deposit_from_coinbase.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+timestamp = str(int(time.time()))
+method = 'POST'
+
+profile_id = 'profile_id'
```
Comment (rcbgr) [inline rest/exchange_create_deposit_from_coinbase.py:27]: Should these be env vars?

File: rest/exchange_create_withdrawal.py [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+# limitations under the License.
+
+import json, hmac, hashlib, time, requests, base64, os, sys
+from urllib.parse import urlparse
+
+API_KEY = str(os.environ.get('ACCESS_KEY'))
+PASSPHRASE = str(os.environ.get('PASSPHRASE'))
+SECRET_KEY = str(os.environ.get('SIGNING_KEY'))
+
+url = f'https://api.exchange.coinbase.com/withdrawals/crypto'
```
Comment (rcbgr) [inline rest/exchange_create_withdrawal.py:22]: No string interpolation. Check the test

File: rest/exchange_create_withdrawal_to_coinbase.py [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+API_KEY = str(os.environ.get('ACCESS_KEY'))
+PASSPHRASE = str(os.environ.get('PASSPHRASE'))
+SECRET_KEY = str(os.environ.get('SIGNING_KEY'))
+
+url = f'https://api.coinbase.com/withdrawals/coinbase-account'
+
+timestamp = str(int(time.time()))
+method = 'POST'
+
+profile_id = 'profile_id'
```
Comment (rcbgr) [inline rest/exchange_create_withdrawal_to_coinbase.py:27]: Env vars?

File: rest/exchange_get_order_by_id.py [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+
+import json, hmac, hashlib, time, requests, base64, os, sys
+from urllib.parse import urlparse
+
+API_KEY = str(os.environ.get('ACCESS_KEY'))
+PASSPHRASE = str(os.environ.get('PASSPHRASE'))
+SECRET_KEY = str(os.environ.get('SIGNING_KEY'))
+
+if len(sys.argv) != 2:
+    print("Error: One command-line argument is required.")
```
Comment (rcbgr) [inline rest/exchange_get_order_by_id.py:23]: Same change as above

**Review by rcbgr — CHANGES_REQUESTED** [review]

File: rest/exchange_create_deposit_from_coinbase.py [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+SECRET_KEY = str(os.environ.get('SIGNING_KEY'))
+PROFILE_ID = str(os.environ.get('PROFILE_ID'))
+
+url = 'https://api.exchange.coinbase.com/deposits/coinbase-account'
+
+timestamp = str(int(time.time()))
+method = 'POST'
+
+amount = 'amount'
+coinbase_account_id = 'coinbase_account_id'
```
Comment (rcbgr) [inline rest/exchange_create_deposit_from_coinbase.py:29]: Does this need to be an env var?

File: rest/exchange_create_withdrawal.py [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+PROFILE_ID = str(os.environ.get('PROFILE_ID'))
+
+url = 'https://api.exchange.coinbase.com/withdrawals/crypto'
+
+timestamp = str(int(time.time()))
+method = 'POST'
+
+amount = '1'
+currency = 'USDC'
+crypto_address = 'crypto_address'
```
Comment (rcbgr) [inline rest/exchange_create_withdrawal.py:30]: Does this need to be an env var or CLI arg?

File: rest/exchange_create_withdrawal_to_coinbase.py [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+SECRET_KEY = str(os.environ.get('SIGNING_KEY'))
+PROFILE_ID = str(os.environ.get('PROFILE_ID'))
+
+url = 'https://api.coinbase.com/withdrawals/coinbase-account'
+
+timestamp = str(int(time.time()))
+method = 'POST'
+
+amount = 'amount'
+coinbase_account_id = 'coinbase_account_id'
```
Comment (rcbgr) [inline rest/exchange_create_withdrawal_to_coinbase.py:29]: Same with the rest of these

**Review by rcbgr — CHANGES_REQUESTED** [review]

**Review by rcbgr — APPROVED** [review]

**Review by cfluke-cb — APPROVED** [review]

File: rest/exchange_create_deposit_from_coinbase.py (hunk 10 above)
Comment (CarolinaMomma843) [reply rest/exchange_create_deposit_from_coinbase.py:29]: Idk what that means I'm new

File: rest/exchange_create_withdrawal.py (hunk 11 above)
Comment (CarolinaMomma843) [reply rest/exchange_create_withdrawal.py]: Idk what that means I'm new

**Review by CarolinaMomma843 — APPROVED** [review]

**Pull request merged**